### PR TITLE
[fluentd-elasticsearch] Make index_name changable

### PR DESCRIPTION
**Pull request by sbibow**
#### What this PR does / why we need it:
A new feature to have a configurable [index_name](https://docs.fluentd.org/output/elasticsearch#index_name-optional).

This should enable people using this chart to configure the index_name to their liking and requirements.
Before the PR it was a choice between a large `fluentd` index or a daily `<logstashPrefix>-%Y.%m.%d` index.

#### Which issue this PR fixes
  - N/A

#### Special notes for your reviewer:
N/A

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://developercertificate.org) signed
- [x] Chart Version bumped (if the pr is an update to an existing chart)
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[fluentd-elasticsearch]`)

### CODE REVIEW

**Comment by nlarzon** [comment]
I was just having a similar problem. I was under the impression u could set the logstash prefix to expand the k8s namespace.
logstash:
    enabled: true
    prefix: "${record['kubernetes']['namespace_name']}"

But I can't get it to work.. it results in the index ${record['kubernetes']['namespace_name']}-2020.05.19

**Comment by rgarcia89** [comment]
@nlarzon @sbibow 
works for me
```
  logstash:
    enabled: true
    prefix: "${record['kubernetes']['namespace_name']}-ams1"
  outputType: "elasticsearch_dynamic"
```

Have you changed the output type to `elasticsearch_dynamic`?

**Comment by nlarzon** [comment]
No, I haven't will try that thanks!

**Comment by monotek** [comment]
Is this a breaking change for people updating from 9.0.0 to 9.1.0?

**Comment by sbibow** [comment]
@monotek Nope.

My variable `elasticsearch.indexName` has the same default value `"fluentd"` as the elasticsearch output module (see <https://docs.fluentd.org/output/elasticsearch#index_name-optional>)

**Comment by monotek** [comment]
thanks! :)

please resolve merge conflicts...

File: charts/fluentd-elasticsearch/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: fluentd-elasticsearch
-version: 9.0.1
+version: 9.1.1
```
Comment (sbibow) [inline charts/fluentd-elasticsearch/Chart.yaml:3]: Shouldn't it have been 9.1.0? The patch version would be reset after a minor update, right?

Well, now it's too late...

Comment (monotek) [reply charts/fluentd-elasticsearch/Chart.yaml:3]: damn, should be 9.0.2.
but we'll leave it this way, as its already published.

**Pull request merged**